.github/workflows/merge-to-stage.js

```diff
@@ -130,6 +132,7 @@ const merge = async ({ prs, type }) => {
 
   for await (const { number, files, html_url, title } of prs) {
     try {
+      if (mergeLimitExceeded()) return
```

Comment:
Couldn't we exit even earlier in the `merge` function?

Reply:
I considered this but it needs to be precisely here for this case:
- We check a batch it has 6 PRs
- We have 3 PRs that are ready to be merged
- We merge 2 PRs within this loop
- We are at 8 now
- Last PR is skipped due to exceeding the limit